docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -0,0 +1,216 @@
+---
+title: Unit testing and mocking with the Azure SDK for .NET
+description: Learn techniques and tools for unit testing and mocking the Azure SDK for .NET
+ms.custom: devx-track-dotnet, engagement-fy23
+ms.date: 07/05/2023
+---
+
+# Unit testing and mocking with the Azure SDK for .NET
+
+Unit testing is an important part of a sustainable development process that can improve code quality and prevent regressions or bugs in your apps. However, unit testing presents challenges when the code you're testing performs network calls to Azure resources. Tests that run against live services can experience issues, such as latency that slows down test execution, dependencies on code outside of the isolated test, and issues with managing service state and costs every time the test is run. Instead of testing against live Azure services, replace the service clients with mocked or in-memory implementations. This avoids the above issues and lets developers focus on testing their application logic, independent from the network and service.
```

Comment:
```suggestion
Unit testing is an important part of a sustainable development process that can improve code quality and prevent regressions or bugs in your apps. However, unit testing presents challenges when the code you're testing performs network calls, such as those made to Azure resources. Tests that run against live services can experience issues, such as latency that slows down test execution, dependencies on code outside of the isolated test, and issues with managing service state and costs every time the test is run. Instead of testing against live Azure services, replace the service clients with mocked or in-memory implementations. This avoids the above issues and lets developers focus on testing their application logic, independent from the network and service.
```